tests/generation/test_modeling.py

```diff
@@ -12,22 +12,32 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-import tempfile
 import time
 import unittest
+from pathlib import Path
+from tempfile import TemporaryDirectory
 
 import torch
+from packaging.version import parse
 from parameterized import parameterized
-from transformers import AutoModelForCausalLM, AutoTokenizer, PretrainedConfig, pipeline, set_seed
+from transformers import (
+    AutoModelForCausalLM,
+    AutoModelForSeq2SeqLM,
+    AutoTokenizer,
+    PretrainedConfig,
+    pipeline,
+    set_seed,
+)
 
-from optimum.intel.generation.modeling import TSModelForCausalLM
+from optimum.intel.generation.modeling import TSModelForCausalLM, TSModelForSeq2SeqLM
 
 
 MODEL_NAMES = {
     "bloom": "hf-internal-testing/tiny-random-BloomModel",
     "gptj": "hf-internal-testing/tiny-random-gptj",
     "gpt2": "hf-internal-testing/tiny-random-gpt2",
     "gpt_neo": "hf-internal-testing/tiny-random-GPTNeoModel",
+    "t5": "google/flan-t5-base",
```

Comment:
We should use a tiny model so that the tests are not too computationally expensive
```suggestion
    "t5": "hf-internal-testing/tiny-random-t5",
```